qiime/sdk/visualization.py

```diff
@@ -33,3 +35,7 @@ def _from_data_dir(cls, data_dir, provenance):
         # shutil.copytree doesn't allow the destination directory to exist.
         viz._archiver.save_data(data_dir, distutils.dir_util.copy_tree)
         return viz
+
+    def get_indices(self):
+        indices = glob.glob(os.path.join(self._archiver._data_dir, 'index.*'))
```

Comment:
`_data_dir` may not exist (Artifacts/Visualizations) are lazy and won't self-extract until needed. Also I'm not a fan of accessing the private functionality of a instance member. The `Archiver` should probably provide some public method for getting the data-directory (allowing it to be extracted/populated if needed).